department-of-veterans-affairs/caseflow
# Lianna/APPEALS -10818: System Check for VHA Business Line Membership

Resolves [APPEALS-10818](https://vajira.max.gov/browse/APPEALS-10818)

### Description
User story development work will be for both higher level review and supplemental claim workflows

### Acceptance Criteria
- [ ] 1. When in taking one of the following decision reviews, a system check will be performed to determine user membership in the Veterans Health Administration organization (business line organization) in Caseflow 
- [ ] 1. Higher Level Review (HLR) 
- [ ] 2. Supplemental claim (SC)


### Testing Plan
1. After downloading this branch open the rails console in Terminal using the `make c` command
2. Once in rails console, run the following command: `bauer =  User.find_by_css_id('ACBAUERVVHAH')`
![Bauer](https://user-images.githubusercontent.com/17394371/202204714-1bba82b9-c7a4-4fb5-8558-e033d4fb064e.png)

3. To verify that this user is a member of the VHA Organizations run `bauer.organizations`
![Bauer Orgs](https://user-images.githubusercontent.com/17394371/202205222-27ac52c0-6e48-4c87-899e-33bef6c75412.png)

4. Once user is confirmed to be a member of the VHA Organization run the new method on the user:  `bauer.vha_employee?`
5. This should return `true`
![Bauer VHA Check](https://user-images.githubusercontent.com/17394371/202205338-d4732f4f-4391-4ee1-8d58-033efd25a88d.png)

6. Next check a user that is not part of the VHA organization following similar commands as steps 2-4. 
7. Run `ian =  User.find_by_css_id('MAILINTAKE_LOCAL')`
![Ian](https://user-images.githubusercontent.com/17394371/202206585-fbdc6620-f523-4b5a-a322-024bd8950c7d.png)

8. Then run `ian.organizations` to confirm user is NOT and member of the VHA organization, (this may result in an empty array) signifying user is not part of any organization.
![Ian Orgs](https://user-images.githubusercontent.com/17394371/202206722-fb4b970c-683c-467e-b625-b23141388d6a.png)

9. Run the new method on this user` ian.vha_employee?`
10. This should return `false`
![Ian VHA Check](https://user-images.githubusercontent.com/17394371/202206814-1a669cae-a93f-4529-a61f-382e0ff7bcc2.png)

11. Since our second user (ian) didn't have any organizations run with one more user
12. Run: `adwise = adwise = User.find_by_css_id('BVADWISE')`
13. Then run: `adwise.organizations`
14. Lastly run:`adwise.vha_employee?`
15. Result should be false

![adwise check all](https://user-images.githubusercontent.com/17394371/202206948-a60e9dc0-16db-4aea-ae8a-bf52b2422543.png)

16. 2 RSpec Tests were created for this new user method. Check that these tests are passing by exiting the rails console and running: 
`make one-test ./spec/models/user_spec.rb:747`

![Spec Tests 10818](https://user-images.githubusercontent.com/17394371/202206984-6f0bdfd7-3c79-4003-a91b-aa88e7e9d586.png)
